app/scripts/lib/nonce-tracker.js

```diff
@@ -83,12 +78,77 @@ class NonceTracker {
     return mutex
   }
 
+  async _getNetworkNonceAndDetails (address) {
+    // calculate next nonce
+    // we need to make sure our base count
+    // and pending count are from the same block
+    const currentBlock = await this._getCurrentBlock()
+    const blockNumber = currentBlock.blockNumber
+    const baseCountHex = await this._getTxCount(address, currentBlock)
+    const baseCount = parseInt(baseCountHex, 16)
+    assert(Number.isInteger(baseCount), `nonce-tracker - baseCount is not an integer - got: (${typeof baseCount}) "${baseCount}"`)
+    // if the nonce provided by the network is higher then a pending tx
+    // toss out the pending txCount
+    const networkNonce = baseCount
+
+    return {networkNonce, blockNumber, baseCountHex, baseCount}
```

Comment:
networkNonce is now only baseCount?
also networkNonce and baseCount both included in nonce details